torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -713,6 +759,9 @@ def _post_backward_hook(
             if (
                 not _low_precision_hook_enabled(state)
                 and flat_param.grad.dtype != handle._reduce_dtype
+                # If we are forcing full precision but communicating grads
```

Comment:
I am a bit confused by the "but communicating grads" part of the comment. IIUC, at this point, we are already certainly communicating gradients, as this is the pre-collective downcast.

Is the statement just meaning like "if we are forcing full precision, don't downcast gradient"?